src/components/App/ImportStudyList.js

```diff
@@ -0,0 +1,168 @@
+import React, { PureComponent } from 'react';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    TextField,
+} from '@material-ui/core';
+import { queryWebsoc } from '../../helpers';
+import RightPaneStore from '../../stores/RightPaneStore';
+import { addCourse, openSnackbar } from '../../actions/AppStoreActions';
+import AppStore from '../../stores/AppStore';
+import { GetApp } from '@material-ui/icons';
+import { termData } from '../../termData';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+
+class ImportStudyList extends PureComponent {
+    state = {
+        isOpen: false,
+        selectedTerm: RightPaneStore.getFormData().term,
+        studyListText: '',
+    };
+
+    handleChange = (event) => {
+        this.setState({ selectedTerm: event.target.value });
+    };
+
+    handleError = (error) => {
+        openSnackbar('error', 'An error occurred while trying to import the Study List.');
+        console.error(error);
+    };
+
+    handleOpen = () => {
+        this.setState({ isOpen: true });
+    };
+
+    handleClose = async (doImport) => {
+        this.setState({ isOpen: false }, () => {
+            document.removeEventListener('keydown', this.enterEvent, false);
+            this.setState({ studyListText: '' });
+        });
+        if (doImport) {
+            document.removeEventListener('keydown', this.enterEvent, false);
+            if (!this.state.studyListText.match(/\d{5}/g)) {
+                openSnackbar('error', 'Cannot import an empty/invalid Study List.');
+                return;
+            }
+            const currSchedule = AppStore.getCurrentScheduleIndex();
+            const sectionCodes = this.state.studyListText
+                .split('\n')
+                .map((line) => line.match(/\d{5}/g))
+                .filter((id) => id)
+                .flat();
+            let sectionsAdded = 0;
+            try {
+                (
+                    await Promise.all(
+                        sectionCodes
+                            .reduce((result, item, index) => {
+                                // WebSOC queries can have a maximum of 8 course codes in tandem
```

Comment:
Where did you find this info or how did you figure this out? I tried changing the chunk size to higher numbers when testing your code and it still seems to work.

Reply:
devsdevsdevs mentioned it [here](https://github.com/icssc-projects/AntAlmanac/pull/302#discussion_r827536026) and it seemed to be correct based on initial testing; I did some more testing with the REST API and it seems like the limit is 10. Thanks for pointing that out—I'll address it.